pyam/core.py

```diff
@@ -392,20 +393,60 @@ def validate(self, criteria={}, exclude=False):
         """
         df = _apply_criteria(self.data, criteria, in_range=False)
 
-        if exclude:
-            idx = _meta_idx(df)
-            self.meta.loc[idx, 'exclude'] = True
-
         if not df.empty:
             msg = '{} of {} data points to not satisfy the criteria'
             logger().info(msg.format(len(df), len(self.data)))
 
-            if exclude and len(idx) > 0:
-                logger().info('{} non-valid scenario{} will be excluded'
-                              .format(len(idx), '' if len(idx) == 1 else 's'))
+            if exclude:
+                self._exclude(df)
 
             return df
 
+    def check_aggregate(self, var_total, var_sector=None, units=None,
+                        exclude=False, threshold=0.1, multiplier=1):
+        """Check whether the timeseries values match the aggregation
+        of sub-categories
+
+        Parameters
+        ----------
+        var_total: str
+            variable to be checked for matching aggregation of sub-categories
+        var_sector: list of str, default None
+            list of variables, defaults to all sub-categories of `var_total`
+        threshold: number, default 0.1
+            threshold for identifiying aggregate as `non-matching`
+            (to allow rounding errors)
+        multiplier: number, default 1
+            factor when comparing var_total and sum of var_sector
+        """
+        # default var_sector to all variables one level below `var_total`
+        if var_sector is None:
+            var_sector = self.filter({'variable': '{}|*'.format(var_total),
+                                      'level': 0}).variables()
+
+        df_total = _aggregate_by_variables(self.data, var_total, units)
+        df_sector = _aggregate_by_variables(self.data, var_sector, units)
+        n = len(df_total)
+
+        diff = df_total - multiplier * df_sector
+        diff = diff[abs(diff) > threshold]
```

Comment:
perhaps you could use here

```
np.isclose(df_total, df_sector, **kwargs)
```

and then allow tolerances and such in the kwargs

see https://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.isclose.html